File: 곽승규/Week_9/P150370_개인정보수집유효기간.py
def solution(today, terms, privacies):
    answer = []

    # 오늘 날짜정보를 일로 변경
    tmp = today.split(".")
    sumDay = (int(tmp[0]) * 28 * 12) + (int(tmp[1]) * 28) + int(tmp[2])

    # 약관 종류 딕셔너리 저장
    term = {}
    for i in terms:
        temp = i.split()
        term[temp[0]] = temp[1]

    cnt = 1
    for i in privacies:
        date, type = i.split()
        tmp = date.split(".")
        # 날짜정보를 일로 변경
        sumD = (int(tmp[0]) * 28 * 12) + (int(tmp[1]) * 28) + int(tmp[2])
        # 비교하기
        if (int(term[type]) * 28) <= abs(sumD - sumDay):
            answer.append(cnt)
        cnt += 1

    return answer

File: 신선영/Week_09/P150370_개인정보수집유효기간.js
function solution(today, terms, privacies) {
    var answer = [];
    
    // 오늘 날짜를 숫자로 변경
    let d = today.split(".")
    let cnt = Number(d[0] * 28 * 12) + Number(d[1] * 28) + Number(d[2])
    let i = 0;
    
    for (let i = 0; i < privacies.length; i++) {
        let p = privacies[i];
        const [date, type] = p.split(" ")
        for (let t of terms) {
            if (t.split(" ")[0] == type) {
                let dd = date.split(".")
                // 기준 날짜를 숫자로 바꿔서 오늘과 비교
                let spent = Number(dd[0] * 28 * 12) + Number(dd[1] * 28) + Number(dd[2])
                // 조건별 유효기간과 오늘과의 차이를 비교해서 지났으면 인덱스값 answer에 저장
                if (Math.abs(spent - cnt) >= (t.split(" ")[1] * 28)) {
                    answer.push(i + 1);
                }

            }
        }
    }
    return answer;
}
